File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    int MOD = 1e9 + 7;
    int com(int n, int r) {
        long long res = 1;
        for(int i = n; i > n - r; i--) {
            res *= i;
            if(n - i + 1 <= r) res /= n - i + 1;
        }
        return res;
    }
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> v;
        for(int i = 0; i < numRows; i++) {
            vector<int> a(i + 1);
            v.push_back(a);
            for(int j = 0; j < i + 1; j++) {
                v[i][j] = com(i, j);
            }
        }
        return v;
    }
};

File: 118-pascals-triangle/pascals-triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        IList<IList<int>> li = new List<IList<int>>(numRows);
        for(int i = 0; i < numRows; i++) {
            IList<int> v = new List<int>(i + 1);
            for(int j = 0; j <= i; j++) {
                if(j == 0 || j == i) {
                    v.Add(1);
                } else {
                    v.Add(li[i - 1][j] + li[i - 1][j - 1]);
                }
            }
            li.Add(v);
        }
        return li;
    }
}